src/transformers/generation/utils.py

```diff
@@ -1477,7 +1479,7 @@ def _get_cache(self, cache_implementation: str, max_batch_size: int, max_cache_l
                 "config": self.config,
                 "max_batch_size": max_batch_size,
                 "max_cache_len": max_cache_len,
-                "device": self.device,
+                "device": device,
```

Comment:
I dropped this line in the end-to-end PR, in a rebase 😢 . `self.device` can't be called at compilation time